### ISSUE TITLE
Warn if "if" conditions are unintentionally always true or always false 

### ISSUE THREAD

**Issue by CoderJoshDK** [astral-sh/ty#2304]
ty does not emit a warning or error when doing a logical comparison where there is no overlap

https://play.ty.dev/beedfd79-499c-443f-a083-4c874e4b5350
```py
foo: str = "bar"

if foo is not None:
    a = "hit"
    print("hit")

if foo is None:
    a = "miss"
    print("miss")

reveal_type(a)
```

basedpyright does provide `1. Condition will always evaluate to True since the types "Literal['bar']" and "None" have no overlap [reportUnnecessaryComparison]`

mypy does not report anything and same with pyright

Another interesting quirk from the playground is that ty knows that the second condition check is unreachable and will then report a as
`1. Revealed type: `Literal["hit"]` [revealed-type]`
But funny enough, basedpyright reports it as `1. Type of "a" is "Literal['miss', 'hit']"`

---

Emitting a warning when there is no overlap between checks feels like an amazing rule to have! 

Chain that brought this issue on https://github.com/rotki/rotki/pull/11269/ and https://discord.com/channels/1039017663004942429/1279201882337705994/1456681723470418123

**Comment by AlexWaygood** [comment]
Interesting! I would have expected mypy to flag the first condition with `--strict-equality --enable-error-code=redundant-expr`, but even with those flags added it doesn't complain about it. (Mypy _does_ complain about the second condition if you add `--warn-unreachable`.)

**Comment by AlexWaygood** [comment]
This is related to https://github.com/astral-sh/ty/issues/576

**Comment by carljm** [comment]
I think it's also related to #1948

**Comment by charliermarsh** [comment]
These are now reported by `redundant-condition-strict`.
